### PR TITLE
Future Stop Market Order in Out Market Hours

### LINKED ISSUE #6948: Future Fill Model Should Fill Stop Market Order Outside Market Hours
#### Expected Behavior
Futures fill model fills stop market orders outside market hours.

#### Actual Behavior
Futures fill model doesn't fill stop market orders outside market hours. it inherits this behavior from the `FillModel`.

#### Potential Solution
Implement `StopMarketFill` in the `FutureFillModel` to override the default behavior.

#### Reproducing the Problem
```csharp
namespace QuantConnect.Algorithm.CSharp
{
    public class GeekyYellowKangaroo : QCAlgorithm
    {
        private OrderTicket _ticket;
        public override void Initialize()
        {
            SetStartDate(2023, 2, 1);
            
            var future = AddFuture(Futures.Indices.NASDAQ100EMini, 
                extendedMarketHours:true);

            Schedule.On(DateRules.Today, TimeRules.At(19,0),
            () => 
            {
                MarketOrder(future.Mapped, 1);
                _ticket = StopMarketOrder(future.Mapped, -1, future.Price*0.999m);
            });
        }

        public override void OnData(Slice slice)
        {
            if (_ticket == null || _ticket.Status != OrderStatus.Submitted) return;
            var stopPrice = _ticket.Get(OrderField.StopPrice);
            var bar = Securities[_ticket.Symbol].Cache.GetData<TradeBar>();
            if (stopPrice > bar.Low) Log($"{stopPrice} -> {bar.Low}");
        }
    }
}
```

The log will show that the low is lower than the stop price, but 7 pm is extended market hours. The order is filled when the market opens (RTH) on Feb 6th.

#### Checklist
- [x] I have completely filled out this template
- [x] I have confirmed that this issue exists on the current `master` branch
- [x] I have confirmed that this is not a duplicate issue by searching [issues](https://github.com/QuantConnect/Lean/issues)
- [x] I have provided detailed steps to reproduce the issue

### PR DESCRIPTION
**Pull request by Romazes**
#### Description

Add opportunity for Future securities to create `StopMarketOrder` in out market hour.
#### Related Issue

Closes #6948
#### Motivation and Context

I will want to help users to create new algorithms with new option of Future security.

#### Requires Documentation Change

N/A
#### How Has This Been Tested?

- unit test
- regression algo tests on `.cs` | `.py`
#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [x] My code follows the code style of this project.
- [x] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [x] I have added tests to cover my changes. 
- [x] All new and existing tests passed.
- [x] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

File: Tests/Common/Orders/Fills/FutureFillModelTests.cs [hunk 1]
```diff
@@ -64,6 +64,49 @@ public void Setup()
...
+            security.SetMarketPrice(new IndicatorDataPoint(symbol, time, marketPrice));
+
+            var fill = model.Fill(new FillModelParameters(
+                security,
+                order,
+                new MockSubscriptionDataConfigProvider(config),
+                Time.OneHour,
+                null)).Single();
+
+            if (extendedMarketHours && isInternal != true)
```
Comment (jhonabreul) [inline Tests/Common/Orders/Fills/FutureFillModelTests.cs:94]: `extendedMarketHours && !isInternal`?

File: Algorithm.CSharp/FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        {
+            SetStartDate(2013, 10, 6);
+            SetEndDate(2013, 10, 12);
+
+            var future = AddFuture(Futures.Indices.SP500EMini, Resolution.Minute, extendedMarketHours: true);
+
+            Schedule.On(DateRules.EveryDay(), TimeRules.At(19, 0), () =>
+            {
+                MarketOrder(future.Mapped, 1);
+                _ticket = StopMarketOrder(future.Mapped, -1, future.Price * 0.999m);
```
Comment (jhonabreul) [inline Algorithm.CSharp/FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm.cs:44]: Should assert that this order is filled on extended hours, perhaps in `OnOrderEvent`

**Review by jhonabreul — CHANGES_REQUESTED** [review]
Thank you! Leaving some minor comments.

File: Tests/Common/Orders/Fills/FutureFillModelTests.cs (hunk 1 above)
Comment (Romazes) [reply Tests/Common/Orders/Fills/FutureFillModelTests.cs]: I have removed `isInternal` flag. It is not make sense.

File: Algorithm.CSharp/FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm.cs (hunk 2 above)
Comment (Romazes) [reply Algorithm.CSharp/FutureStopMarketOrderOnExtendedHoursRegressionAlgorithm.cs]: Added:
- UTest
- regression algos

**Review by jhonabreul — CHANGES_REQUESTED** [review]
Cool! Minor comments

**Review by jhonabreul — APPROVED** [review]
Good work! Looks good 💯

**Review by Martin-Molinero — APPROVED** [review]
Looks good ty👍 , minor comments shared through slack

**Review by Martin-Molinero — APPROVED** [review]

**Pull request merged**